R/api-functions.R

```diff
@@ -0,0 +1,79 @@
+# Data package identifier reservation ------------------------------------------
+#' Reserve EDI ID 
+#' @description This package returns a funder
+#' @param user_id A string that contains the user ID for the EDI data portal. If you 
+#' do not already have an EDI account you may create one on the EDI portal. \url {https://portal.edirepository.org/nis/login.jsp}
+#' @param password A string that contains the user password for the EDI data portal. 
+#' @return This function returns a edi identifier number. 
+#' @examples 
+#' reserve_edi_id(user_id = "samuelwright", 
+#'                password = "340account")
+#' @export                
+
+reserve_edi_id <- function(user_id, password) {
+  r <-httr::POST(
+    url = "https://pasta-d.lternet.edu/package/reservations/eml/edi",
+    config = httr::authenticate(paste('uid=', user_id, ",o=EDI", ',dc=edirepository,dc=org'), password)
+  )
+  if (r$status_code == "201") {
+  edi_number <- httr::content(r, as = 'text', encoding = 'UTF-8')
+  paste0("edi.", edi_number, ".1", sep = "")
+} else {
+  message("Your request to reserve an EDI number failed, 
+          please check that you entered a valid username and password. 
+          See more information on request status below")
+  print(r)
+}
+}
+
+# Evaluate EDI Data package -------------------------------------------------------
+#' Evaluate EDI Package 
+#' @description This function takes in authentication info for edi and an EML file to 
+#' be evaluated using the EDI congruency checker. This package returns a data frame that contains the status of the 
+#' package. 
+#' @param user_id A string that contains the user ID for the EDI data portal. If you 
+#' do not already have an EDI account you may create one on the EDI portal. \url {https://portal.edirepository.org/nis/login.jsp}
+#' @param password A string that contains the user password for the EDI data portal. 
+#' @param eml_file_path A string that contains the filepath to the EML document that 
+#' contains all the data package information. A web link to the csv must be included 
+#' in the dataset information in the EML in order for a data package to be evaluated. 
+#' @return This package returns a data frame that contains the status of the 
+#' package. The data frame contains a column with the status, a column with the descriotion 
+#' of what each status is referring to, and a column that contains a suggestion of how 
+#' to fix the status or improve the package. 
+#' @examples 
+#' reserve_edi_id(user_id = "samuelwright", 
+#'                password = "340account",
```

Comment:
maybe not .Renviron for examples, but make sure you are doing it for the edi packages you are doing and git ignore the .Renviron file

Reply:
look at how other api wrapper R packages deal with the password in examples. it looks like `rnoaa` just ignores putting a token in the [examples](https://github.com/ropensci/rnoaa/blob/master/R/ncdc.r)